src/schema/objects/formats.yaml

```
@@ -0,0 +1,115 @@
+---
+# This file defines valid patterns for different formats
+# Entity patterns
+label:
+  name: Label
+  description: |
+    Freeform labels without special characters.
+  pattern: '[0-9a-zA-Z]+'
+index:
+  name: Index
+  description: |
+    Non-negative, non-zero integers, optionally prefixed with leading zeros for sortability.
+    An index may not be all zeros.
+  pattern: '[0-9]*[1-9]+[0-9]*'
+# Metadata types
+string:
+  name: String
+  description: |
+    The basic string type (not a specific format).
+    This should allow any free-form string.
+  pattern: '.*'
+integer:
+  name: Integer
+  description: |
+    An integer which may be positive or negative.
+  pattern: '[+-]?\d+'
+number:
+  name: Number
+  description: |
+    A number which may be an integer or float, positive or negative.
+  pattern: '[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)'
+boolean:
+  name: Boolean
+  description: |
+    A boolean.
+    Must be either "true" or "false".
+  pattern: '(true|false)'
+# String formats
+date:
+  name: Date
+  description: |
+    A date in the form `"YYYY-MM-DD[Z]"`,
+    where [Z] is an optional, valid timezone code.
+  pattern: '[0-9]{4}-[0-9]{2}-[0-9]{2}?([A-Z]{3,4})'
+datetime:
+  name: Datetime
+  description: |
+    A datetime in the form `"YYYY-MM-DDThh:mm:ss[.000000][Z]"`,
+    where [.000000] is an optional subsecond resolution between 1 and 6 decimal points,
+    and [Z] is an optional, valid timezone code.
+  pattern: '[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}?(\.[0-9]{1,6})?([A-Z]{3,4})'
+time:
+  name: Time
+  description: |
+    A time in the form `"hh:mm:ss"`.
+  pattern: '(?:2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9]'
```

Should we permit `1:00:00`? Or just `01:00:00`?

```suggestion
  pattern: '(?:2[0-3]|[01]?[0-9]):[0-5][0-9]:[0-5][0-9]'
```

---

I think that's reasonable. Thanks!